test/FastTests/TestsInheritanceFastTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace FastTests
+{
+    public class TestsInheritanceFastTests : RavenLowLevelTestBase
+    {
+        public TestsInheritanceFastTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [NonLinuxFact]
+        public void AllTestsShouldUseRavenFactOrRavenTheoryAttributes()
+        {
+            var assemblies = new HashSet<Assembly>();
+            string projectPath = FindSlowTestsDirectory();
+            string outputDir = NewDataPath(suffix: "BuildOutput"); // Temporary directory
+            Directory.CreateDirectory(outputDir);
+
+            var projectCsProj = Path.Combine(projectPath, "SlowTests.csproj");
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "dotnet",
+                    Arguments = $"publish \"{projectCsProj}\" --configuration Debug --output \"{outputDir}\" --no-restore",
```

Comment:
why not `dotnet build` ?

Reply:
we are building inside running project, if I use dotnet build, then it will try to copy dependencies and we will get "file in use exception" but if I specify output folder and use publish then it includes all necessary dependencies, preventing missing assembly errors and process cannot access the file exceptions.